src/runtime/lpython_builtin.py

```diff
@@ -1018,16 +1016,50 @@ def _lpython_str_isascii(s: str) -> bool:
             return False
     return True
 
-def _lpython_str_isspace(s:str) -> bool:
+def _lpython_str_isspace(s: str) -> bool:
+    # A Unicode character is considered a 'whitespace' if it has has a bidirectional
+    # type 'WS', 'B' or 'S'; or the category 'Zs'.
+    WHITESPACE: list[str] = [
+        " ",        # SPACE
+        "\n",       # LINE FEED (LF)
+        "\r",       # CARRIAGE RETURN (CR)
+        "\t",       # CHARACTER TABULATION (HT)
+        "\v",       # VERTICAL TAB (VT)
+        "\f",       # FORM FEED (FF)
+        "\u00A0",   # NO-BREAK SPACE
+        "\u1680",   # OGHAM SPACE MARK
+        "\u2000",   # EN QUAD
+        "\u2001",   # EM QUAD
+        "\u2002",   # EN SPACE
+        "\u2003",   # EM SPACE
+        "\u2004",   # THREE-PER-EM SPACE
+        "\u2005",   # FOUR-PER-EM SPACE
+        "\u2006",   # SIX-PER-EM SPACE
+        "\u2007",   # FIGURE SPACE
+        "\u2008",   # PUNCTUATION SPACE
+        "\u2009",   # THIN SPACE
+        "\u200A",   # HAIR SPACE
+        "\u2028",   # LINE SEPARATOR
+        "\u2029",   # PARAGRAPH SEPARATOR
+        "\u202F",   # NARROW NO-BREAK SPACE
+        "\u205F",   # MEDIUM MATHEMATICAL SPACE
+        "\u3000",   # IDEOGRAPHIC SPACE
+    ]
     if len(s) == 0:
         return False
-    ch: str 
-    for ch in s:
-        if ch != ' ' and ch != '\t' and ch != '\n' and ch != '\r' and ch != '\f' and ch != '\v':
+    
+    char: str
+    w_char: str
+    for char in s:
+        is_whitespace: bool = False
+        for w_char in WHITESPACE:
+            if char == w_char:
+                is_whitespace = True
+                break
+        if not is_whitespace:
```

Comment:
I thought `if char in WHITESPACE: ...` works, but it is not the case.
So, Let's keep it as the following for now.
```suggestion
        if not (char == " "  or   # SPACE
            char == "\n"     or   # LINE FEED (LF)
            char == "\r"     or   # CARRIAGE RETURN (CR)
            char == "\t"     or   # CHARACTER TABULATION (HT)
            char == "\v"     or   # VERTICAL TAB (VT)
            char == "\f"     or   # FORM FEED (FF)
            char == "\u00A0" or   # NO-BREAK SPACE
            char == "\u1680" or   # OGHAM SPACE MARK
            char == "\u2000" or   # EN QUAD
            char == "\u2001" or   # EM QUAD
            char == "\u2002" or   # EN SPACE
            char == "\u2003" or   # EM SPACE
            char == "\u2004" or   # THREE-PER-EM SPACE
            char == "\u2005" or   # FOUR-PER-EM SPACE
            char == "\u2006" or   # SIX-PER-EM SPACE
            char == "\u2007" or   # FIGURE SPACE
            char == "\u2008" or   # PUNCTUATION SPACE
            char == "\u2009" or   # THIN SPACE
            char == "\u200A" or   # HAIR SPACE
            char == "\u2028" or   # LINE SEPARATOR
            char == "\u2029" or   # PARAGRAPH SEPARATOR
            char == "\u202F" or   # NARROW NO-BREAK SPACE
            char == "\u205F" or   # MEDIUM MATHEMATICAL SPACE
            char == "\u3000"      # IDEOGRAPHIC SPACE
        ):
```

Reply:
Strange, but this works fine on my branch. 
```python
from lpython import i32, i64, f32, f64, dict, list, tuple, str, Const, c64

s: str = " \n\f\v\r\t"
print(s.isspace())

l: list[i32] = [1, 2, 3, 4, 5]
i: i32
for i in l:
    print(i)
```
```console
(lp) saurabh-kumar@Awadh:~/Projects/System/lpython$ ./src/bin/lpython ./examples/example.py
True
1
2
3
4
5
```

Reply:
I meant we shouldn't be using a for loop here:
```py
		for w_char in WHITESPACE:
            if char == w_char:
                is_whitespace = True
                break
```

Reply:
Could not get you @Thirumalai-Shaktivel.